debezium-server/src/main/java/io/debezium/server/Server.java

```diff
@@ -0,0 +1,188 @@
+ *
+ * @author Jiri Pechanec
+ *
+ */
+@ApplicationScoped
+@Startup
+public class Server {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(Server.class);
+
+    private static final String PROP_PREFIX = "debezium.";
+    private static final String PROP_CONSUMER_NAME = PROP_PREFIX + "consumer";
+    private static final String PROP_KEY_FORMAT = PROP_PREFIX + "format.key";
+    private static final String PROP_VALUE_FORMAT = PROP_PREFIX + "format.value";
+    private static final String PROP_TERMINATION_WAIT = PROP_PREFIX + "termination.wait";
+
+    private static final String FORMAT_JSON = Json.class.getSimpleName().toLowerCase();
+    private static final String FORMAT_AVRO = Avro.class.getSimpleName().toLowerCase();
+
+    private ExecutorService executor = Executors.newSingleThreadExecutor();
+
+    @Inject
+    BeanManager beanManager;
+
+    @Inject
+    @Liveness
+    ConnectorLifecycle health;
+
+    private Bean<DebeziumEngine.ChangeConsumer<ChangeEvent<?, ?>>> consumerBean;
+    private CreationalContext<ChangeConsumer<ChangeEvent<?, ?>>> consumerBeanCreationalContext;
+    private DebeziumEngine.ChangeConsumer<ChangeEvent<?, ?>> consumer;
+    private DebeziumEngine<?> engine;
+
+    @SuppressWarnings("unchecked")
+    @PostConstruct
+    public void start() {
+        final Config config = ConfigProvider.getConfig();
+        final String name = config.getValue(PROP_CONSUMER_NAME, String.class);
+
+        final Set<Bean<?>> beans = beanManager.getBeans(name).stream()
+                .filter(x -> DebeziumEngine.ChangeConsumer.class.isAssignableFrom(x.getBeanClass()))
+                .collect(Collectors.toSet());
+        LOGGER.debug("Found {} candidate consumer(s)", beans.size());
+
+        if (beans.size() == 0) {
+            throw new DebeziumException("No Debezium consumer named '" + name + "' is available");
+        }
+        else if (beans.size() > 1) {
+            throw new DebeziumException("Multiple Debezium consumers named '" + name + "' were found");
+        }
+
+        consumerBean = (Bean<DebeziumEngine.ChangeConsumer<ChangeEvent<?, ?>>>) beans.iterator().next();
+        consumerBeanCreationalContext = beanManager.createCreationalContext(consumerBean);
+        consumer = consumerBean.create(consumerBeanCreationalContext);
+        LOGGER.info("Consumer '{}' instantiated", consumer.getClass().getName());
+
+        final Class<? extends SerializationFormat<?>> keyFormat = getFormat(config, PROP_KEY_FORMAT);
+        final Class<? extends SerializationFormat<?>> valueFormat = getFormat(config, PROP_VALUE_FORMAT);
+        final Properties props = configToProperties(config);
+        props.setProperty("name", name);
+
+        DebeziumEngine.Builder<?> builder = null;
+        // TODO - apply variance and covariance rules on Debezium API to
+        // support direct assignment to DebeziumEngine.Builder<ChangeEvent<?, ?>>
+        if (keyFormat == Json.class && valueFormat == Json.class) {
```

Comment:
We really need to get other types enabled (like `Integer` for keys), but that can be done separately.

Reply:
Yes, but for that we need first to solve to generification of Debezium API using covariance and contravariance